include/zephyr/linker/common-ram.ld

```diff
@@ -151,3 +151,13 @@ ITERABLE_SECTION_RAM(scmi_protocol, Z_LINK_ITERABLE_SUBALIGN)
 #ifdef CONFIG_USERSPACE
 	PLACE_SYMBOL_HERE(_static_kernel_objects_end);
 #endif
+
+#if defined(CONFIG_DCACHE_LINE_SIZE)
+	SECTION_DATA_PROLOGUE(data_cache,,SUBALIGN(CONFIG_DCACHE_LINE_SIZE))
+	{
+		PLACE_SYMBOL_HERE(__data_cache_start);
+		KEEP(*(SORT_BY_NAME(".data_cache*")))
+		. = ALIGN(CONFIG_DCACHE_LINE_SIZE);
+		PLACE_SYMBOL_HERE(__data_cache_end);
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+#endif /* CONFIG_DCACHE_LINE_SIZE */
```

Comment:
I don't like that this automatically means, that all variables with this attribute land in the data section. That means that variables, are initialized with zero, will no longer land in the bss section. That leads to increase in size in flash, as they now can't be optimized out.